### PR TITLE
NN: start and stop nav session

### PR DESCRIPTION
**Pull request by RingerJK**
### Description
NN: start and stop nav session

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 1]
```diff
@@ -1992,8 +1992,11 @@ class MapboxNavigation @VisibleForTesting internal constructor(
             navigatorConfig,
         )
         historyRecorderHandles = createHistoryRecorderHandles(config)
+
+        val navSessionState = navigator.restoreNavigationSession()
```
Comment (dzinad) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:1996]: NIT: can we do this inside recreate?

File: libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt [hunk 2]
```diff
@@ -127,6 +130,17 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
...
 
+    override fun startNavigationSession() {
+        navigator!!.startNavigationSession()
+    }
+
+    override fun stopNavigationSession() {
+        navigator!!.stopNavigationSession()
+    }
+
+    override fun restoreNavigationSession(): NavigationSessionState =
```
Comment (dzinad) [inline libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt:141]: Maybe `storeNavigationSession`?

**Review by dzinad — APPROVED** [review]

File: libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigator.kt [hunk 3]
```diff
@@ -58,6 +60,12 @@ interface MapboxNativeNavigator {
 
     suspend fun resetRideSession()
 
+    fun startNavigationSession()
+
+    fun stopNavigationSession()
+
+    fun restoreNavigationSession(): NavigationSessionState
```
Comment (averkhaturau) [inline libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigator.kt:67]: How store / restore parity is implemented?

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt (hunk 1 above)
Comment (RingerJK) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt]: make sense, done

File: libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigator.kt (hunk 3 above)
Comment (RingerJK) [reply libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigator.kt]: discussed, moved store/restore under impl

**Review by averkhaturau — APPROVED** [review]
lgtm

**Pull request merged**